friendly-requests-client/src/containers/FaqPage.js

```diff
@@ -0,0 +1,55 @@
+import React, { Component } from 'react'
+import { Accordion, Container, Icon } from 'semantic-ui-react'
+
+export default class FaqPage extends Component {
+  state = { activeIndex: 0 }
+
+  handleClick = (e, titleProps) => {
+    const { index } = titleProps
+    const { activeIndex } = this.state
+    const newIndex = activeIndex === index ? -1 : index
+
+    this.setState({ activeIndex: newIndex })
+  }
```

Comment:
It's often nice to have more semantically meaningful callback names. In this case we don't really know what's happening in here until we dig in and compile it in our heads. This one even is apparently special and related to `<Accordion>` in the arguments it passes.

One possible option…

```js
toggleAccordion = ( e, { index: clicked } ) => {
	const { activeIndex: active } = this.state;

	// if we clicked the open item then close them all
	// but if we clicked on another one then open that one
	this.setState( { activeIndex: active === clicked ? null : clicked } );
}
```

Reply:
I also tossed in the `null` vs the `-1` just as a way to indicate "this isn't supposed to represent a valid selected index" but `-1` is also fine